app/models/user.rb

```diff
@@ -3,8 +3,25 @@ class User < ActiveRecord::Base
   has_many :tags, as: :subscriber
   has_many :questions
   has_many :answers
+  has_many :social_providers
 
-  validates :uuid, presence: true
-  validates :provider, presence: true
+  def self.from_omniauth(auth, user=nil)
+    user = where(email: auth.info.email).first_or_create do |u|
+      u.name= auth.info.name
+      u.email= auth.info.email
+    end
+    user.social_providers.from_social(auth)
+    user
+  end
+
+  def refresh_token
+    token = TokenManager.generate_token(self.id)
```

Comment:
nope. but it would be useful if we want to attach more processes to the `refresh_token` method